packages/client/src/lib.js

```diff
@@ -94,15 +104,40 @@ class NFTStorage {
   }
   /**
    * @param {API.Service} service
-   * @param {Blob} blob
-   * @returns {Promise<API.CIDString>}
+   * @param {Blob|CarReader} carReader
+   * @returns {Promise<API.CIDString | undefined>}
    */
-  static async storeCar({ endpoint, token }, blob) {
-    const car =
-      blob.type !== 'application/car'
-        ? blob.slice(0, blob.size, 'application/car') // TODO: test this in safari.
-        : blob
-    return NFTStorage.storeBlob({ endpoint, token }, car)
+  static async storeCar({ endpoint, token }, carReader) {
```

Comment:
Option for `onProgress`?

Reply:
This would be nice, but I am not sure we have enough data here. We do not know how many chunks will exist on the fly, nor the total number of bytes